packages/lesswrong/components/bookmarks/ReadHistoryTab.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {useState} from 'react';
+import { registerComponent, Components, fragmentTextForQuery } from '../../lib/vulcan-lib';
+import {AnalyticsContext} from "../../lib/analyticsEvents";
+import {useCurrentUser} from "../common/withUser"
+import { gql, useQuery, NetworkStatus } from '@apollo/client';
+import moment from 'moment';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  loadMore: {
+    marginTop: 10
+  },
+  loadMoreSpinner: {
+    textAlign: 'left',
+    paddingTop: 6,
+    paddingLeft: 10,
+    margin: 0
+  }
+})
+
+const ReadHistoryTab = ({classes}: {classes: ClassesType}) => {
+  const currentUser = useCurrentUser()
+  const defaultLimit = 10;
+  const pageSize = 30;
+  const [limit, setLimit] = useState(defaultLimit);
+  
+  // pull the latest 10 posts that the current user has read
+  const { data, loading, fetchMore, networkStatus } = useQuery(gql`
+    query getReadHistory($limit: Int) {
+      UserReadHistory(limit: $limit) {
+        posts {
+          ...PostsListWithVotes
+          lastVisitedAt
+        }
+      }
+    }
+    ${fragmentTextForQuery("PostsListWithVotes")}
+    `,
+    {
+      ssr: true,
+      fetchPolicy: "cache-and-network",
```

Comment:
Not super important but I noticed that this query never seems to be cached - if you navigate between the tabs then the other two will be display imeditely the second time you visit them, but this one always loads.

Probably not worth spending a ton of time on. I'd guess it probably predates this PR too.